data-runtime/src/main/java/io/micronaut/data/runtime/mapper/sql/SqlResultEntityTypeMapper.java

```diff
@@ -622,6 +632,11 @@ private <K> K triggerPostLoad(RuntimePersistentEntity<?> persistentEntity, K ent
     private <K> Object readEntityId(RS rs, MappingContext<K> ctx) {
         RuntimePersistentProperty<K> identity = ctx.persistentEntity.getIdentity();
         if (identity == null) {
+            // DTO might not have ID mapped, and in this case to maintain relation
+            // we set random UUID as id to be able to read and make relation
+            if (isDto) {
+                return UUID.randomUUID();
```

Comment:
Are you sure this will work?

Reply:
Yes, there is a test that covers that case.

Reply:
On the other hand, don't know if could break some user case, all tests we have are passing.